pkg/virt-launcher/virtwrap/cli/libvirt.go

```diff
@@ -343,6 +346,82 @@ func (l *LibvirtConnection) GetDomainStats(statsTypes libvirt.DomainStatsTypes,
 	return list, nil
 }
 
+func (l *LibvirtConnection) GetDomainDirtyRate(calculationDuration time.Duration, flags libvirt.DomainDirtyRateCalcFlags) ([]*stats.DomainStatsDirtyRate, error) {
```

Comment:
I think we need to lock this function. I'm not sure what happens if you call StartDirtyRateCalc multiple times... 
What do you think @iholder101 ?

Reply:
> I'm not sure what happens if you call StartDirtyRateCalc multiple times...

Yeah I'm not sure as well :)
However, I think we already handle this problem. The cache is defined with `useValueLock=true` which means that the `Get()` call (hence the recalc function) can only be executed by one entity at a time:
https://github.com/kubevirt/kubevirt/blob/d0dd376266d025ff02e612a1cc733526904dc77d/tools/cache/time-defined-cache.go#L64-L68